opentelemetry-config/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+//! # Library for declarative configuration of OpenTelemetry.
+//!
+//! This library provides a way to configure OpenTelemetry SDK components
+//! using a declarative approach. It allows users to define configurations
+//! for metrics, traces, and exporters in a structured manner.
+
+use std::{
+    collections::HashMap,
+    error::{self, Error},
+    fmt::{self, Display},
+};
+
+use opentelemetry_sdk::{
+    logs::SdkLoggerProvider,
+    metrics::{MeterProviderBuilder, SdkMeterProvider},
+    trace::SdkTracerProvider,
+};
+use serde_yaml::Value;
+
+pub mod model;
+pub mod providers;
+
+/// Registry for different configuration providers.
+pub struct ConfigurationProvidersRegistry {
+    metrics: MetricsProvidersRegistry,
+}
+
+impl ConfigurationProvidersRegistry {
+    pub fn new() -> Self {
+        Self {
+            metrics: MetricsProvidersRegistry::new(),
+        }
+    }
+
+    pub fn metrics_mut(&mut self) -> &mut MetricsProvidersRegistry {
+        &mut self.metrics
+    }
+
+    pub fn metrics(&self) -> &MetricsProvidersRegistry {
+        &self.metrics
+    }
+}
+
+impl Default for ConfigurationProvidersRegistry {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// Registry for metrics configuration providers.
+pub struct MetricsProvidersRegistry {
+    periodic_exporter_factories: HashMap<String, Box<MetricConfigFactory>>,
+    // TODO: Add other types of providers registries.
+}
+
+impl MetricsProvidersRegistry {
+    pub fn new() -> Self {
+        Self {
+            periodic_exporter_factories: HashMap::new(),
+        }
+    }
+
+    pub fn register_periodic_exporter_factory(
```

Comment:
For the reasons mentioned in https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/487#discussion_r2532499511

```suggestion
    pub fn register_metrics_config(
```